### PR TITLE
Bump GSL to 2.3

### PR DESCRIPTION
**Pull request by dsendkowski**

### CODE REVIEW

**Comment by dsendkowski** [comment]
How is it that this component's FMRI is `library/math/gsl` but IPS has it under `developer/gsl` ?

**Comment by alarcher** [comment]
Because a different FMRI was hardcoded in the manifest here:
https://github.com/OpenIndiana/oi-userland/pull/3163/files#diff-7c9156ff1ccdc1f7fcec396f1c5f455aL26

I mentioned it last time but the PR was merged before it was fixed.
To me developer/gsl does not make sense.

File: components/library/gsl/Makefile [hunk 1]
```diff
@@ -14,35 +14,34 @@
...
-COMPONENT_ARCHIVE_URL	= ftp://ftp.gnu.org/gnu/gsl/$(COMPONENT_ARCHIVE)
-COMPONENT_FMRI			= library/math/gsl
+COMPONENT_NAME=           gsl
+COMPONENT_VERSION=        2.3
+COMPONENT_SUMMARY=        GSL - GNU Scientific Library
+COMPONENT_SRC=            $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=        $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=   sha256:562500b789cd599b3a4f88547a7a3280538ab2ff4939504c8b4ac4ca25feadfb
+COMPONENT_ARCHIVE_URL=    ftp://ftp.gnu.org/gnu/gsl/$(COMPONENT_ARCHIVE)
+COMPONENT_FMRI=           library/math/gsl
```
Comment (pyhalov) [inline components/library/gsl/Makefile]: Please, fix variables, affecting package meta-information, to match what they currently set to.

**Review by pyhalov — CHANGES_REQUESTED** [review]

File: components/library/gsl/Makefile [hunk 2]
```diff
@@ -14,35 +14,34 @@
...
-COMPONENT_ARCHIVE_URL	= ftp://ftp.gnu.org/gnu/gsl/$(COMPONENT_ARCHIVE)
-COMPONENT_FMRI			= library/math/gsl
+COMPONENT_NAME=           gsl
+COMPONENT_VERSION=        2.3
+COMPONENT_SUMMARY=        GSL - GNU Scientific Library
+COMPONENT_SRC=            $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=        $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=   sha256:562500b789cd599b3a4f88547a7a3280538ab2ff4939504c8b4ac4ca25feadfb
+COMPONENT_ARCHIVE_URL=    ftp://ftp.gnu.org/gnu/$(COMPONENT_NAME)/$(COMPONENT_ARCHIVE)
+COMPONENT_FMRI=           library/math/gsl
```
Comment (alarcher) [inline components/library/gsl/Makefile]: You just need to change this to developer/gsl as Alexander specified.

**Comment by dsendkowski** [comment]
I thought you wanted to change it. I can see only one component (`inkscape`) depending on this component explicitly. Anyway, as you wish.

**Pull request merged**